app/src/main/java/com/example/planetsapptraining/ui/fragments/PlanetDetailFragment.kt

```diff
@@ -21,11 +30,13 @@ class PlanetDetailFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_planet_detail, container, false)
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        val planet = arguments.planet
-        textPlanetName.text = planet.name
-        textPlanetShortDesc.text = planet.shortDescription
-        Glide.with(view).load(planet.imageUrl).into(imagePlanet)
+    private fun render(viewState: PlanetViewState) {
+        textPlanetName.text = viewState.name
+        textPlanetShortDesc.text = viewState.shortDescription
+        Glide.with(this).load(viewState.imageUrl).into(imagePlanet)
+        view_detail_row_distance_from_sun.render("Distance from sun", viewState.distanceFromSun.toString())
+        view_detail_row_surface_gravity.render("Surface gravity", viewState.surfaceGravity.toString())
+        view_detail_row_planet_type.render("Planet type", viewState.planetType.toString())
+        view_detail_row_description.render("Description", viewState.description.toString())
```

Comment:
Not sure this is the best of doing it. Thought of using a recycler view or reflection, but ended up doing like this not to fall in the "overkill" pitfall.

Reply:
I like it!